src/makeCalculator.js

```diff
@@ -1,43 +1,31 @@
 'use strict';
 
-/**
- * Another calculator. Now the task is more difficult.
- * Create a `makeCalculator` function that returns an object that
- * has the following fields:
- *  - Methods: `add`, `subtract`, `multiply`, `divide`, `reset`, `operate`.
- *  - The `result` property is initially 0.
- *
- * How the calculator will work:
- * - Each `operate` call takes a callback and a number and sets the
- *   appropriate value to the `result` property.
- * - The `reset` method resets `result` value to 0.
- * - `add`, `subtract`, `multiply`, `divide` are passed as callbacks to
- *   `operate` method
- * - The `operate` and `reset` methods can be called in a chain.
- *
- * Example:
- * const calculator = makeCalculator();
- *
- * calculator.operate(calculator.add, 21)
- * calculator.result === 21
+function makeCalculator() {
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
 
- * calculator.reset()
- * calculator.result === 0
+      return this;
+    },
+    add(result, value) {
+      return result + value;
+    },
+    subtract(result, value) {
+      return result - value;
+    },
+    multiply(result, value) {
+      return result * value;
+    },
+    divide(result, value) {
+      return result / value;
+    },
```

Comment:
You need to add a check for 0, since we cannot divide by 0 according to the rules of mathematics